README.md

```diff
@@ -3,4 +3,42 @@
 A Discord bot with useful utilities for the
 [`dart_community` Discord server][discord].
 
+## Features
+
+### `/ping`
+
+The `ping` command checks the bot is online and displays some latency
+metrics.
+
+### `/show-lint`
+
+The `show-lint` command displays information about a given linter rule from
+https://dart.dev/tools/linter-rules.
+
+### Dartdoc search
+
+`dash_bot` will parse messages for references to API elements or packages from
+https://pub.dev and provide a link to that element.
+
+Searches can be embedded in any text message and their syntax is as follows:
+- `![Name]` or `![package/Name]`: Return the documentation for `Name` in
+  Flutter's or `package`'s API documentation.
+- `?[Name]` or `?[package/Name]`: Search for `Name` in Flutter's or
+  `package`'s API documentation.
+- `$[name]`: Return the pub.dev page for the package `name`.
+- `&[name]`: Search pub.dev for `name`.
+
+## Running dash_bot
+
+If you can use `dart:mirrors` (i.e during development), simply set the `TOKEN`
```

Comment:
`nyxx_commands` uses `dart:mirrors` if it is available, since the built script to get the metadata needed runs quite slowly. It's quicker to use mirrors if it's available, and then only use the built script once we're building an executable.

As for what it actually uses mirrors for, `nyxx_commands` allows you to specify command arguments by adding them as arguments to the callback function (see the `show-lint` command and how it takes an argument of type `LintRule`). It needs to fetch the argument metadata for all command callbacks, which it can either do using mirrors or using the build script (though it isn't `build_runner` compatible).